src/libraries/System.Data.Common/src/System/Data/Common/DbDataSource.cs

```diff
@@ -0,0 +1,570 @@
+                }
+            }
+
+            protected override async Task<DbDataReader> ExecuteDbDataReaderAsync(
+                CommandBehavior behavior,
+                CancellationToken cancellationToken)
+            {
+                await _connection.OpenAsync(cancellationToken).ConfigureAwait(false);
+
+                try
+                {
+                    return await _wrappedCommand.ExecuteReaderAsync(
+                            behavior | CommandBehavior.CloseConnection,
+                            cancellationToken)
+                        .ConfigureAwait(false);
+                }
+                catch
+                {
+                    try
+                    {
+                        await _connection.CloseAsync().ConfigureAwait(false);
+                    }
+                    catch (Exception e)
+                    {
+                        ExceptionBuilder.TraceExceptionWithoutRethrow(e);
+
+                        // Swallow to allow the original exception to bubble up
+                    }
+
+                    throw;
+                }
+            }
+
+            protected override DbParameter CreateDbParameter()
+                => _wrappedCommand.CreateParameter();
+
+            public override void Cancel()
+                => _wrappedCommand.Cancel();
+
+            public override void Prepare()
+                => _wrappedCommand.Prepare();
+
+            public override Task PrepareAsync(CancellationToken cancellationToken = default)
+                => _wrappedCommand.PrepareAsync(cancellationToken);
+
+            [AllowNull]
+            public override string CommandText
+            {
+                get => _wrappedCommand.CommandText;
+                set => _wrappedCommand.CommandText = value;
+            }
+
+            public override int CommandTimeout
+            {
+                get => _wrappedCommand.CommandTimeout;
+                set => _wrappedCommand.CommandTimeout = value;
+            }
+
+            public override CommandType CommandType
+            {
+                get => _wrappedCommand.CommandType;
+                set => _wrappedCommand.CommandType = value;
+            }
+
+            protected override DbParameterCollection DbParameterCollection
+                => _wrappedCommand.Parameters;
+
+            public override bool DesignTimeVisible
+            {
+                get => _wrappedCommand.DesignTimeVisible;
+                set => _wrappedCommand.DesignTimeVisible = value;
+            }
+
+            public override UpdateRowSource UpdatedRowSource
+            {
+                get => _wrappedCommand.UpdatedRowSource;
+                set => _wrappedCommand.UpdatedRowSource = value;
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                var connection = _wrappedCommand.Connection;
+
+                _wrappedCommand.Dispose();
+                connection!.Dispose();
```

Comment:
These lines should be in an `if (disposing) { ... }` block. (`disposing == false` means either (a) being called from a finalizer, in which case managed objects shouldn't be accessed, or (b) it's called from `DisposeAsync` in which case the resources have already been disposed via `DisposeAsyncCore` so this is duplicative.)

Reply:
Yep, thanks - this was an oversight.